cmake/modules/RootMacros.cmake

```diff
@@ -393,28 +393,39 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
   endforeach()
 
   #---Build the names for library, pcm and rootmap file ----
+  set(library_target_name)
   if(dictionary MATCHES "^G__")
-    string(SUBSTRING ${dictionary} 3 -1 deduced_arg_module)
+    string(REGEX REPLACE "^G__(.*)" "\\1"  library_target_name ${dictionary})
+    if (ARG_MULTIDICT)
+      string(REGEX REPLACE "(.*)32$" "\\1"  library_target_name ${library_target_name})
```

Comment:
Is this really going to work for stuff outside ROOT? It can be the source of subtle bugs for external users.

Reply:
Nope, this is pretty terrible approach, however, I am covering here the MULTIDICT case a little better. This code slightly more functional from what it was before.

Reply:
Those places are when we have multiple dictionaries. Smatrix and GenVector. This has been the naming convention unfortunately. I am not sure what you want to change. I am all for removing the dictionary names provided by the users (as explained in the other comment)